src/Task/FilePermissionTask.php

```diff
@@ -0,0 +1,131 @@
+        return array(
+            AccompliEvents::INSTALL_RELEASE => array(
+                array('onInstallReleaseUpdateFilePermissions', 0),
+            ),
+        );
+    }
+
+    /**
+     * Constructs a new FilePermissionTask.
+     *
+     * @param array $paths
+     */
+    public function __construct($paths)
+    {
+        $this->paths = $paths;
+    }
+
+    /**
+     * Sets the correct permissions and group for the configured path.
+     *
+     * @param InstallReleaseEvent      $event
+     * @param string                   $eventName
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function onInstallReleaseUpdateFilePermissions(InstallReleaseEvent $event, $eventName, EventDispatcherInterface $eventDispatcher)
+    {
+        $host = $event->getRelease()->getWorkspace()->getHost();
+        $connection = $this->ensureConnection($host);
+
+        $eventDispatcher->dispatch(AccompliEvents::LOG, new LogEvent(LogLevel::INFO, 'Set correct permissions for configured paths...', $eventName, $this, array('event.task.action' => TaskInterface::ACTION_IN_PROGRESS)));
+
+        if ($connection->isConnected() === true && $connection->isDirectory($event->getRelease()->getPath())) {
+            foreach (array_keys($this->paths) as $configuredPath) {
+                $path = $this->getPath($configuredPath, $event);
+                $permissions = $this->getPermissionsForPath($configuredPath);
+                $recursive = $this->getRecursive();
+
+                $result = $connection->changePermissions($path, $permissions, $recursive);
+            }
+
+            if ($result === true) {
+                $eventDispatcher->dispatch(AccompliEvents::LOG, new LogEvent(LogLevel::INFO, 'Permissions set for configured paths.', $eventName, $this, array('event.task.action' => TaskInterface::ACTION_COMPLETED, 'output.resetLine' => true)));
+            } else {
+                $eventDispatcher->dispatch(AccompliEvents::LOG, new LogEvent(LogLevel::WARNING, 'Failed setting the correct permissions for configured paths.', $eventName, $this, array('event.task.action' => TaskInterface::ACTION_FAILED)));
+            }
+        }
+    }
+
+    /**
+     * Gets the directory for setting permissions.
+     *
+     * @param string              $configuredPath
+     * @param InstallReleaseEvent $event
+     *
+     * @return string
+     */
+    protected function getPath($configuredPath, $event)
+    {
+        $releasePath = $event->getRelease()->getPath();
+
+        return $releasePath.'/'.$configuredPath;
+    }
+
+    /**
+     * Gets the calculated permissions based on the configuration.
+     *
+     * @param string $path
+     *
+     * @return int
+     *
+     * @throws TaskCommandExecutionException
+     */
+    protected function getPermissionsForPath($path)
+    {
+        if (array_key_exists('permissions', $this->paths[$path])) {
+            $configuredPermissions = array_column($this->paths, 'permissions')[0];
```

Comment:
`array_column` is not usable in this situation. This line always returns the permissions of the first path...